app/services/arcgis_api/geocoder.rb

```diff
@@ -126,6 +126,8 @@ def handle_api_errors(response_body)
       if response_body['error']
         error_code = response_body.dig('error', 'code')
 
+        retrieve_token! if error_code === 498
```

Comment:
this just retrieves a new token for the next request, do we also want to retry the api (limited number of retrying)

if we reorganized this code to throw an error when we see this error code, we rewrite to catch that error + use ruby's `retry` keyword or the `with_retries` gem that has a limit

this is a very rough outline

```ruby
def list_locations
  response = with_api_token(method: method, headers: headers, body: body)
  JSON.parse(response) 
end

def with_api_token(method:, headers:, body:)
  token = retrieve_token!
  response = faraday.request(method, headers, body)
  if response.code == 498
    retry # TODO: add some counter to make sure no infinite loop
  end
  response
end
```